### PR TITLE
React to breaking changes in Microsoft.AspNetCore.App

### PR DESCRIPTION
**Pull request by natemcmaster**
~~WIP until https://github.com/aspnet/AspNetCore/pull/4004 merges and has a build available.~~

Changes:

* Add packages references for EF Core, when necessary
* Add packages references for SpaServices to Spa templates
* Remove PackageReference to Microsoft.AspNetCore.App altogether

Resolves https://github.com/aspnet/AspNetCore/issues/3612

cc @dsplaisted

### CODE REVIEW

File: src/Microsoft.DotNet.Web.ProjectTemplates/RazorClassLibrary-CSharp.csproj.in [hunk 1]
```diff
@@ -1,11 +1,12 @@
...
 
   <PropertyGroup>
-    <TargetFramework>netstandard2.0</TargetFramework>
+    <TargetFramework>netcoreapp3.0</TargetFramework>
     <RootNamespace Condition="'$(name)' != '$(name{-VALUE-FORMS-}safe_namespace)'">Company.RazorClassLibrary1</RootNamespace>
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.AspNetCore.Mvc" Version="${MicrosoftAspNetCoreMvcPackageVersion}" />
+    <FrameworkReference Include="Microsoft.AspNetCore.App" />
```
Comment (natemcmaster) [inline src/Microsoft.DotNet.Web.ProjectTemplates/RazorClassLibrary-CSharp.csproj.in]: cc @pranavkm @rynowak @dsplaisted - it wasn't clear to me if Microsoft.NET.Sdk.Razor would also imply this framework reference, so I've left it in for now.

Comment (dsplaisted) [reply src/Microsoft.DotNet.Web.ProjectTemplates/RazorClassLibrary-CSharp.csproj.in]: Right now, the Razor SDK doesn't implicitly add the framework reference.  We can discuss whether it should, I don't really know right now.

**Comment by natemcmaster** [comment]
:up: :date: tests are passing with the latest build of aspnetcore.

File: src/Microsoft.DotNet.Web.Spa.ProjectTemplates/ReactRedux-CSharp.csproj.in [hunk 2]
```diff
@@ -12,7 +12,7 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.AspNetCore.App" />
```
Comment (ryanbrandenburg) [inline src/Microsoft.DotNet.Web.Spa.ProjectTemplates/ReactRedux-CSharp.csproj.in]: I guess I'm confused about under what circumstances we need a FrameworkReference? Why do we need it in RazorClassLibrary but not here?

Comment (natemcmaster) [reply src/Microsoft.DotNet.Web.Spa.ProjectTemplates/ReactRedux-CSharp.csproj.in]: The current plan of record is that `Microsoft.NET.Sdk.Web` implies a FrameworkReference to aspnetcore. RazorClassLibrary doesn't use the websdk, it uses Microsoft.NET.Sdk.Razor

**Pull request merged**